src/rust/engine/src/watch.rs

```diff
@@ -49,6 +50,7 @@ impl InvalidationWatcher {
     executor: Executor,
     build_root: PathBuf,
     ignorer: Arc<GitignoreStyleExcludes>,
+    enabled: bool,
```

Comment:
It feels to me that creating a disabled `InvalidationWatcher` is a weird pattern. Would it be possible to replace this logic with the caller holding an `Option<InvalidationWatcher>` and avoid the need for any change to this class?